pkgs/build-support/fetchzip/default.nix

```diff
@@ -5,47 +5,46 @@
 # (e.g. due to minor changes in the compression algorithm, or changes
 # in timestamps).
 
-{ fetchurl, unzip }:
+{ lib, fetchurl, unzip }:
 
-{ # Optionally move the contents of the unpacked tree up one level.
-  stripRoot ? true
+{ name ? "source"
 , url
+  # Optionally move the contents of the unpacked tree up one level.
+, stripRoot ? true
 , extraPostFetch ? ""
-, name ? "source"
 , ... } @ args:
 
 (fetchurl ({
   inherit name;
 
   recursiveHash = true;
-
   downloadToTemp = true;
 
-  postFetch =
-    ''
-      unpackDir="$TMPDIR/unpack"
-      mkdir "$unpackDir"
-      cd "$unpackDir"
+  postFetch = ''
```

Comment:
This should do the same thing without any hash changes or rebuilds, as verified by doing `nix-build -A <something>.src --check` for a bunch of <something>s that use various fetchers.  Same for the fetchgit whitespace indentation cleanup.